web/modules/custom/joinup_communities/tallin/src/Plugin/Field/FieldType/TallinEntryItem.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Drupal\tallin\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\TypedData\MapDataDefinition;
+
+/**
+ * Plugin implementation of the 'tallin_entry' field type.
+ *
+ * @FieldType(
+ *   id = "tallin_entry",
+ *   label = @Translation("Tallin entry"),
+ *   description = @Translation("Stores data about a tallin section."),
+ *   default_widget = "tallin_entry_default",
+ *   default_formatter = "tallin_entry"
+ * )
+ */
+class TallinEntryItem extends FieldItemBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultFieldSettings() {
+    return parent::defaultFieldSettings() + [
+      'tallin_description' => '',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function fieldSettingsForm(array $form, FormStateInterface $form_state) {
+    $element['tallin_description'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Description of the Tallin section'),
+      '#description' => $this->t('The definition of the Tallin agreement section. This will be shown in the form.'),
+      '#default_value' => $this->getSetting('tallin_description'),
+    ];
+    $element += parent::fieldSettingsForm($form, $form_state);
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function propertyDefinitions(FieldStorageDefinitionInterface $field_definition) {
+    // Properties related to the options value.
+    $properties['status'] = DataDefinition::create('string')
+      ->setLabel('Status')
+      ->addConstraint('Length', ['max' => 255])
+      ->setRequired(FALSE);
+
+    // Properties related to the explanation value.
+    $properties['value'] = DataDefinition::create('string')
+      ->setLabel(t('Details'))
+      ->setRequired(FALSE);
+    $properties['format'] = DataDefinition::create('filter_format')
+      ->setLabel(t('Text format'));
+
+    // Properties related to the link value.
+    $properties['uri'] = DataDefinition::create('uri')
+      ->setLabel(t('URI'));
+    $properties['title'] = DataDefinition::create('string')
```

Comment:
I don't see any use-case of `title` and `options` properties. At least the widget and the formatter are not using it.